determined-ai/determined
# chore: Split UserRoles out of Store

## Description

chore: Split UserRoles out of Store [WEB-585]

This splits UserRoles and UserAssignments into their own store. 





## Test Plan

1. Navigate to the `WebUI` open the console and ensure there are no errors related to permissions.
2. Activate the `rbac` feature flag using `?f_rbac=on`. Ensure there is a `501` error due to `/api/v1/permissions/summary` and the UI still loads correctly. 





## Commentary (optional)

I have removed the initialization of the user roles that was: 
```
  userRoles: [
    {
      id: -10,
      name: 'INITIALIZATION',
      permissions: [],
    },
  ],
```

please let me know if that needs to be restored.






## Checklist

~- [ ] User-facing API changes need the "User-facing API Change" label.~
~- [ ] Release notes should be added as a separate file under `docs/release-notes/`.~
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
~- [ ] Licenses should be included for new code which was copied and/or modified from any external code.~




[WEB-585]: https://determinedai.atlassian.net/browse/WEB-585?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
